cli/lib/kontena/cli/stacks/yaml/validator_v3.rb

```diff
@@ -69,6 +69,9 @@ def validate(yaml)
         if yaml.key?('services')
           if yaml['services'].kind_of?(Hash)
             yaml['services'].each do |service, options|
+              unless service =~ /\A(?!-)[a-z0-9\-]+\z/
```

Comment:
Compared to `\w`, this also rejects `A-Z`?

```
- test.yaml:
  - services:
      Redis:
        name: A service name should only include a-z, 0-9 and - characters and not
          start with the - character
```

Reply:
OTOH DNS names are case-insensitive, so this might actually be a server bug if it lets you create two services called `redis` and `Redis`... they will overlap eachother in DNS?

Reply:
Doesn't reject, just warns.

Reply:
It might indeed make sense to warn about uppercase characters in the service name, yes...